src/module-config/module-config.ts

```diff
@@ -16,7 +32,19 @@ export function provide(config) {
   configs.push(config);
 }
 
-export function getConfig(moduleName) {
+export function useModuleConfig() {
```

Comment:
Let's implement this function in a separate PR.

This PR has been open for too long - I'd much prefer a quicker turnaround where we create, review, merge, and then continue doing that.

By increasing the scope of a PR, reviewers have to start from scratch when reviewing this. Myself and a few others wrote more about this topic at my previous job in [this google doc](https://docs.google.com/document/d/1_jCuT3vRSW8WKJbT9manso_VM0JoUIb6HpFmBi_yDMs/edit).

Reply:
Yeah, I'm totally on board with the approach to PRs in your Google doc, it's a great document. There seems to be some tension between the desire to 1. do WIP PRs for features that might require significant back-and-forth on implementation (e.g. this one); 2. having fast turnaround on PRs; and 3. not committing bugs to master. I'll add some comments around this code (as it was before, as `getConfig`) indicating that it isn't actually working yet.